src/main/java/eu/dissco/core/digitalspecimenprocessor/service/ProcessingService.java

```diff
@@ -483,47 +536,103 @@ private void rollBackToEarlierVersion(DigitalSpecimenRecord currentDigitalSpecim
+          digitalSpecimenRecordsWithMediaEr.keySet());
       if (!bulkResponse.errors()) {
-        handleSuccessfulElasticInsert(digitalSpecimenRecords);
+        handleSuccessfulElasticInsert(digitalSpecimenRecordsWithMediaEr);
       } else {
-        handlePartiallyFailedElasticInsert(digitalSpecimenRecords, bulkResponse);
+        handlePartiallyFailedElasticInsert(digitalSpecimenRecordsWithMediaEr, bulkResponse);
+      }
+      log.info("Successfully created {} new digitalSpecimenRecord",
+          digitalSpecimenRecordsWithMediaEr.size());
+      annotationPublisherService.publishAnnotationNewSpecimen(
+          digitalSpecimenRecordsWithMediaEr.keySet());
+      if (!mediaPidMap.isEmpty()) {
+        gatherDigitalMediaObjectForNewRecords(digitalSpecimenRecordsWithMediaEr, mediaPidMap);
       }
-      log.info("Successfully created {} new digitalSpecimenRecord", digitalSpecimenRecords.size());
-      annotationPublisherService.publishAnnotationNewSpecimen(digitalSpecimenRecords.keySet());
-      gatherDigitalMediaObjectForNewRecords(digitalSpecimenRecords);
-      return digitalSpecimenRecords.keySet();
+      return digitalSpecimenRecordsWithMediaEr.keySet();
     } catch (IOException | ElasticsearchException e) {
       log.error("Rolling back, failed to insert records in elastic", e);
-      digitalSpecimenRecords.forEach(this::rollbackNewSpecimen);
-      rollbackHandleCreation(digitalSpecimenRecords.keySet().stream().toList());
+      digitalSpecimenRecordsWithMediaEr.forEach(this::rollbackNewSpecimen);
+      rollbackHandleCreation(digitalSpecimenRecordsWithMediaEr.keySet().stream().toList());
       return Collections.emptySet();
-    }
+    } catch (PidException e) {
+      log.error(
+          "An error has occurred creating PIDs for media objects of specimens {}",
+          digitalSpecimenRecordsWithMediaEr.keySet().stream().map(
+              DigitalSpecimenRecord::id).toList());
+      return digitalSpecimenRecordsWithMediaEr.keySet();
+    }
+  }
+
+  private DigitalSpecimenRecord setMediaEntityRelationship(
+      DigitalMediaProcessResult mediaProcessResult,
+      DigitalSpecimenRecord digitalSpecimenRecord, Map<DigitalMediaKey, String> mediaPidMap) {
+    var newEntityRelationshipStream = mediaProcessResult.newMedia().stream()
+        .map(mediaEvent -> {
+          var digitalMediaKey = new DigitalMediaKey(digitalSpecimenRecord.id(),
+              mediaEvent.digitalMediaObjectWithoutDoi().attributes().getAcAccessURI());
+          return buildEntityRelationship(HAS_MEDIA.getName(), mediaPidMap.get(
+              digitalMediaKey));
+        });
+    var totalErs = Stream.concat(
+        digitalSpecimenRecord.digitalSpecimenWrapper().attributes().getOdsHasEntityRelationship()
+            .stream()
+            .filter(entityRelationship -> !mediaProcessResult.tombstoneMedia()
+                .contains(entityRelationship)), // remove tombstone ERs
```

Comment:
This comment looks like a todo. Maybe increase the comment to full sentences or remove it here

Reply:
clarified